### DISCUSSION TITLE
Sentry Errors "You made a post request to ** but did not provide an 'action' for route routes/$"

### DISCUSSION THREAD

**Discussion by lsbyerley** [epicweb-dev/epic-stack#807] [General]
I've recently seen these errors in sentry and I'm just curious if anyone else is familiar with them. I'm assuming they are hacking attempts but I'm also curious if anything should be done to block these or if they are fine since they aren't handled as a route (just ignored basically)

<img width="854" alt="Screenshot 2024-07-25 at 8 21 34 PM" src="https://github.com/user-attachments/assets/20e65aad-b9f2-4f04-bdc9-85fd1ad6ad61">

**Comment by kiliman** [comment]
Yes, those errors are typically caused by bots trying to attack your site.

If you don't want those cluttering your Sentry logs, you can add the following to _server/utils/monitoring.ts_

```ts
Sentry.init({
  dsn: 'YOUR_SENTRY_DSN',
  integrations: [new Sentry.Integrations.HttpContext()], // Ensure HttpContext is captured
  beforeSend(event) {
    // Check if the event contains an HTTP context with a response status code
    // 405 Method Not Allowed
    if (event.contexts?.response?.status_code === 405) {
      // Return null to drop the event and prevent it from being sent to Sentry
      return null;
    }
    // Otherwise, send the event to Sentry
    return event;
  },
});
```

**Reply by lsbyerley** [reply]
awesome, thanks!
